### DISCUSSION TITLE
Manual formatting

### DISCUSSION THREAD

**Discussion by akiross** [tweag/nickel#647] [Q&A] [answered]
Hello,
is there a way to manually formatting a nickel file, instead of using the LSP?

Would be nice to have a CLI such as `nickel fmt` for that. That could be performed by `nickel-lang-lsp` under the hood.m

A couple of use cases:
 - one uses an editor that does not support LSP,
 - git hooks to enforce proper formatting before commit/pushing/merging.

Thanks!

**Comment by yannham** [comment, marked as answer]
Hello, 

Unfortunately there is no code formatter yet, even when using the LSP. We have been actually discussing this subject no later than last week. Someone from the team is considering working on this, as this is important tooling. But the current parser infrastructure is not that easy to reuse or to adapt to make a formatter, so I can't really give an estimation for now on when a formatter would be likely to land. Of course, this will be done as a component used both for a `nickel fmt` command and for the LSP.

**Reply by akiross** [reply]
Hello, thank you for the answer. As you can tell, I'm not using the LSP yet so I was not aware of the missing formatter component :)
